### PR TITLE
feat: improve `create-rise` styling

### PR DESCRIPTION
**Pull request by TheKohan**
image

Another PR to visually please our beloved users!
* Added icons, used `chalk` to make `create-rise` "pop"
* Added option to skip installing packages (`astro` does this, thought it would be nice if we don't support multiple package managers for now)
* Replaced `inquirer` with `@inquirer/prompts` its successor (with a bit cleaner api and access to `theme`  type) 
* Themed inquirer prompts with `stylePrompt` fnc

Feel free to add your opinion's in this PR.

### CODE REVIEW

File: packages/create-rise/src/cli.ts [hunk 1]
```diff
@@ -4,19 +4,29 @@ import path from 'node:path'
...
     {
       type: 'input',
       name: 'projectName',
       message: 'Project Name',
       default: 'rise-project',
     },
+    {
+      type: 'confirm',
+      name: 'installDeps',
+      message: 'Would you like to install dependencies ?',
```
Comment (grabbou) [inline packages/create-rise/src/cli.ts:27]: nit: space before ?

File: packages/create-rise/src/cli.ts [hunk 2]
```diff
@@ -48,7 +58,10 @@ async function createRise() {
       },
     ])
     if (!overwrite) {
-      console.error(`The directory "${root}" already exists. Exiting...`)
+      console.error(dedent`
+        ${chalk.red.bold('🚨 There was an error setting up new Rise project:')}
+        ${chalk.red(`The directory "${root}" already exists. Exiting...`)}
+        `)
```
Comment (grabbou) [inline packages/create-rise/src/cli.ts:64]: nit: make sure to run ESLint on save. Line 64 looks like bad indentation which should've been fixed by ESLint already!

File: packages/create-rise/src/cli.ts [hunk 3]
```diff
@@ -77,14 +90,29 @@ async function createRise() {
...
+    console.log(chalk.bold(`📦 Using ${chalk.bold.cyan('npm')} to install packages.`))
+
+    await spinner(chalk.bold(`🔧 Installing dependencies in ${projectName}`), () => $`npm install`)
+  }
+
+  console.log(riseGradient(WELCOME_TO_RISE_ASCII))
 
   console.log(
-    `The project has been successfully created in ${projectName}. To start, run 'npm dev'`
+    [
```
Comment (grabbou) [inline packages/create-rise/src/cli.ts:107]: nit: we can use `decent` and multiline string, e.g.
```
welcome

to

rise
```
instead of the array notation!

**Review by grabbou — APPROVED** [review]
Love it! Thanks for working on it!

To be honest, I would only add a configuration/theme where we keep primary/secondary colors and use them instead of default green/red for success/failure!

File: packages/create-rise/src/cli.ts [hunk 4]
```diff
@@ -96,11 +124,13 @@ async function copyAdditionalTemplateFiles(root: string) {
 
 createRise().catch((e) => {
```
Comment (grabbou) [inline packages/create-rise/src/cli.ts:125]: Meanwhile @TheKohan while we wait for @ericvicenti thoughts on the logo, I have another quick one for you!

We should switch $`npm install` to be $({ quiet: !verbose })`npm install` and use `minimist` from `zx` to check for that flag and allow passing `--verbose`.

`npm install` generates excessive output and will make the beautiful UI of the CLI you created much uglier!

**Review by grabbou — APPROVED** [review]

**Comment by arpitBhalla** [comment]
Can get pkg manager using `npm_execpath` env var

```ts
const packageManager = () => {
  const execPath = process.env.npm_execpath;

  if (execPath.includes('yarn')) {
    return 'yarn';
  } else if (execPath.includes('pnpm')) {
    return 'pnpm';
  } else {
    return 'npm';
  }
};
```

**Comment by TheKohan** [comment]
> Can get pkg manager using `npm_execpath` env var
> 
> ```ts
> const packageManager = () => {
>   const execPath = process.env.npm_execpath;
> 
>   if (execPath.includes('yarn')) {
>     return 'yarn';
>   } else if (execPath.includes('pnpm')) {
>     return 'pnpm';
>   } else {
>     return 'npm';
>   }
> };
> ```

We wanted to wait and try to port that logic from `react-native-cli`, there's been some discussion going on discord

**Comment by arpitBhalla** [comment]
Got it. I saw npm install in the code and thought we would use npm i until the PR is merged into that repo.

File: packages/create-rise/src/cli.ts (hunk 3 above)
Comment (TheKohan) [reply packages/create-rise/src/cli.ts:107]: You meant `dedent` ? I've tried it but didn't know how to conditionally display `installDeps` without making empty line when the value is not there.

**Comment by grabbou** [comment]
> Got it. I saw npm install in the code and thought we would use npm i until the PR is merged into that repo.

Yes, that is the idea. Until the PR in the React Native Community is created/merged, we're just using `npm`. We may consider adding `--no-install` flag at some point down the road, as a temporary solution, but I don't think it's a problem for now!

File: packages/create-rise/src/cli.ts (hunk 1 above)
Comment (TheKohan) [reply packages/create-rise/src/cli.ts]: done

File: packages/create-rise/src/cli.ts (hunk 2 above)
Comment (TheKohan) [reply packages/create-rise/src/cli.ts]: Weird, it didn't pick it up as an error. My `vscode` should pick the `.vscode/settings.json` by default i think 🤔

**Comment by TheKohan** [comment]
@grabbou  @ericvicenti  I've updated the look 😄

**Pull request merged**